src/query/storage/m3/cluster_resolver_test.go

```diff
@@ -111,3 +114,281 @@ func TestGraphitePath(t *testing.T) {
+	},
+	{
+		name:        "unagg optimized enabled short range",
+		queryLength: time.Minute,
+		opts: &storage.FanoutOptions{
+			FanoutUnaggregated:        storage.FanoutForceEnable,
+			FanoutAggregated:          storage.FanoutForceDisable,
+			FanoutAggregatedOptimized: storage.FanoutForceEnable,
+		},
+		expectedType:         namespaceCoversAllQueryRange,
+		expectedClusterNames: []string{"UNAGG"},
+	},
+	{
+		name: "agg enabled",
+		opts: &storage.FanoutOptions{
+			FanoutUnaggregated:        storage.FanoutForceDisable,
+			FanoutAggregated:          storage.FanoutForceEnable,
+			FanoutAggregatedOptimized: storage.FanoutForceDisable,
+		},
+		expectedType: namespaceCoversPartialQueryRange,
+		expectedClusterNames: []string{"AGG_FILTERED", "AGG_NO_FILTER",
+			"AGG_FILTERED_COMPLETE", "AGG_NO_FILTER_COMPLETE"},
+	},
+	{
+		name:        "agg enabled short range",
+		queryLength: time.Minute,
+		opts: &storage.FanoutOptions{
+			FanoutUnaggregated:        storage.FanoutForceDisable,
+			FanoutAggregated:          storage.FanoutForceEnable,
+			FanoutAggregatedOptimized: storage.FanoutForceDisable,
+		},
+		expectedType: namespaceCoversAllQueryRange,
+		expectedClusterNames: []string{"AGG_FILTERED", "AGG_NO_FILTER",
+			"AGG_FILTERED_COMPLETE", "AGG_NO_FILTER_COMPLETE"},
+	},
+	{
+		name: "unagg and agg enabled",
+		opts: &storage.FanoutOptions{
+			FanoutUnaggregated:        storage.FanoutForceEnable,
+			FanoutAggregated:          storage.FanoutForceEnable,
+			FanoutAggregatedOptimized: storage.FanoutForceDisable,
+		},
+		expectedType: namespaceCoversPartialQueryRange,
+		expectedClusterNames: []string{"UNAGG", "AGG_FILTERED", "AGG_NO_FILTER",
+			"AGG_FILTERED_COMPLETE", "AGG_NO_FILTER_COMPLETE"},
+	},
+	{
+		name:        "unagg and agg enabled short range",
+		queryLength: time.Minute,
+		opts: &storage.FanoutOptions{
+			FanoutUnaggregated:        storage.FanoutForceEnable,
+			FanoutAggregated:          storage.FanoutForceEnable,
+			FanoutAggregatedOptimized: storage.FanoutForceDisable,
+		},
+		expectedType:         namespaceCoversAllQueryRange,
+		expectedClusterNames: []string{"UNAGG"},
+	},
+	{
+		name: "agg and optimization enabled",
+		opts: &storage.FanoutOptions{
+			FanoutUnaggregated:        storage.FanoutForceDisable,
+			FanoutAggregated:          storage.FanoutForceEnable,
+			FanoutAggregatedOptimized: storage.FanoutForceEnable,
+		},
+		expectedType:         namespaceCoversAllQueryRange,
+		expectedClusterNames: []string{"AGG_NO_FILTER", "AGG_NO_FILTER_COMPLETE"},
+	},
+	{
+		name:        "all enabled short range",
+		queryLength: time.Minute,
+		opts: &storage.FanoutOptions{
+			FanoutUnaggregated:        storage.FanoutForceEnable,
+			FanoutAggregated:          storage.FanoutForceEnable,
+			FanoutAggregatedOptimized: storage.FanoutForceEnable,
+		},
+		expectedType:         namespaceCoversAllQueryRange,
+		expectedClusterNames: []string{"UNAGG"},
+	},
+}
+
+func TestCases(t *testing.T) {
```

Comment:
Nit: what's this actually test? Maybe better to call it `TestResolveClusterNamespacesForQuery`?

Reply:
Hahah oops yeah... I mean, it tests cases, right?